packages/rocketchat-livestream/server/routes.js

```diff
@@ -0,0 +1,48 @@
+import google from 'googleapis';
+const OAuth2 = google.auth.OAuth2;
+
+RocketChat.API.v1.addRoute('livestream/oauth', {
+	get: function functionName() {
+		const clientAuth = new OAuth2(RocketChat.settings.get('Broadcasting_client_id'), RocketChat.settings.get('Broadcasting_client_secret'), `${ RocketChat.settings.get('Site_Url') }/api/v1/livestream/oauth/callback`);
+		const { userId } = this.queryParams;
+		console.log('userId', userId);
+		const url = clientAuth.generateAuthUrl({
+			access_type: 'offline',
+			scope: ['https://www.googleapis.com/auth/youtube'],
+			state: JSON.stringify({
+				userId
+			})
+		});
+
+		return {
+			statusCode: 302,
+			headers: {
+				Location: url
+			}, body: 'Oauth redirect'
+		};
+	}
+});
+
+RocketChat.API.v1.addRoute('livestream/oauth/callback', {
+	get: function functionName() {
+		const { code, state } = this.queryParams;
+
+		const { userId } = JSON.parse(state);
+
+		const clientAuth = new OAuth2(RocketChat.settings.get('Broadcasting_client_id'), RocketChat.settings.get('Broadcasting_client_secret'), `${ RocketChat.settings.get('Site_Url') }/api/v1/livestream/oauth/callback`);
+
+		const ret = Meteor.wrapAsync(clientAuth.getToken.bind(clientAuth))(code);
+
+		console.log(ret);
```

Comment:
this `console.log` is important to have?

Reply:
Nope, removing.